### PR TITLE
pra_05

### PR DESCRIPTION
**Pull request by Liu1xuan**
@Rinnki97 @AlexZhang21 @Byeclouds

### CODE REVIEW

File: prac_05/wimbledon.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+"""
+CP1404/CP5632 Practical
+Wimbledon
+Estimate: 30 minutes
+Actual:   40 minutes
+"""
+
+import csv
```
Comment (Rinnki97) [inline prac_05/wimbledon.py:8]: module we didn't learn

File: prac_05/wimbledon.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    print()
+    print(f"These {len(countries)} countries have won Wimbledon: ")
+    print(", ".join(countries))
+
+
+def get_data(filename):
+    """Reads a CSV file and returns its data as a list of lists."""
+    with open(filename, "r", encoding="utf-8-sig") as csv_file:
+        csv_reader = csv.reader(csv_file)
+        next(csv_reader)  # skip header row
```
Comment (Rinnki97) [inline prac_05/wimbledon.py:35]: using a statement that we didn't learn from the lecture, but i think the knowledge that been taught in lecture is enough for our practical

File: prac_05/emails.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        emails_to_names[email] = name
+
+        email = input("Email: ")
+
+    # Loop
+    for email, name in emails_to_names.items():
+        print(f"{name} ({email})")
+
+
+email_check()
```
Comment (Rinnki97) [inline prac_05/emails.py:42]: missing main function

File: prac_05/word_occurrences.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        else:
+            word_counts[word] = 1
+
+    #get the length of longest word
+    longest_word_length = max(len(word) for word in word_counts.keys())
+
+    for word, count in sorted(word_counts.items()):
+        print(f"{word:{longest_word_length}} : {count}")
+
+word_count()
```
Comment (Rinnki97) [inline prac_05/word_occurrences.py:29]: miss main function also

File: prac_05/emails.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+def extract_name(email):
+    """
+    Given an email, extract the name of the user.
+    """
+    parts = email.split('@')[0].split('.')
+    return ' '.join([part.title() for part in parts])
+
+
+def email_check():
```
Comment (Byeclouds) [inline prac_05/emails.py:16]: better use docstring

File: prac_05/wimbledon.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
+"""
+CP1404/CP5632 Practical
+Wimbledon
+Estimate: 30 minutes
+Actual:   40 minutes
+"""
+
+import csv
+from collections import Counter
```
Comment (Byeclouds) [inline prac_05/wimbledon.py:9]: we dind't learn this

**Pull request merged**